aten/src/ATen/native/TensorIterator.h

```diff
@@ -195,6 +199,9 @@ struct CAFFE2_API TensorIterator {
       for(int i=0; i < noutputs(); i++) {
         if (dtype(i) != op_dtype(i)) {
           operands_[i].tensor = operands_[i].tensor.to(op_dtype(i));
+          if (operands_[i].buffer.defined()) {
+            operands_[i].buffer.copy_(operands_[i].tensor);
```

Comment:
copy_ already casts dtype, so in this case you probably don't need to call .to() explicitly above

Reply:
hmmm.... OK. It's possible for the consumer of the TensorIterator to return the original result (modified via buffer) or the `iter.output()` which would refer to `operands_[i].tensor`, but I can probably check conditionally and set one to the other.